topics/indexing.md

```diff
@@ -654,35 +654,37 @@ removing the elements which are outside our search box.
 
 Turning this into code is simple. Here is a Ruby example:
 
-    def spacequery(x0,y0,x1,y1,exp)
-        bits=exp*2
-        x_start = x0/(2**exp)
-        x_end = x1/(2**exp)
-        y_start = y0/(2**exp)
-        y_end = y1/(2**exp)
-        (x_start..x_end).each{|x|
-            (y_start..y_end).each{|y|
-                x_range_start = x*(2**exp)
-                x_range_end = x_range_start | ((2**exp)-1)
-                y_range_start = y*(2**exp)
-                y_range_end = y_range_start | ((2**exp)-1)
-                puts "#{x},#{y} x from #{x_range_start} to #{x_range_end}, y from #{y_range_start} to #{y_range_end}"
-
-                # Turn it into interleaved form for ZRANGE query.
-                # We assume we need 9 bits for each integer, so the final
-                # interleaved representation will be 18 bits.
-                xbin = x_range_start.to_s(2).rjust(9,'0')
-                ybin = y_range_start.to_s(2).rjust(9,'0')
-                s = xbin.split("").zip(ybin.split("")).flatten.compact.join("")
-                # Now that we have the start of the range, calculate the end
-                # by replacing the specified number of bits from 0 to 1.
-                e = s[0..-(bits+1)]+("1"*bits)
-                puts "ZRANGE myindex [#{s} [#{e} BYLEX"
-            }
+```ruby
```

Comment:
Have we tested this in a modern Ruby?

Reply:
No. I don't intend to spend time on that. I'd rather delete it, or the entire page. :laughing:

Reply:
Fair. 

I've heard of actions that will test snippets for syntax errors and stuff like this. Might be worth setting up.

With regard to this particular snippet it was written 9 years ago and someone modified it 2 years ago to remove deprecated Ruby stuff according to git blame. That said, I'm fine leaving it as-is, but I think the problem (generally) still exists.

Reply:
OK, let's leave this one as is then.

The interactive `valkey-cli` is very cool and maybe we can integrate it for the valkey-cli command session examples, but I think it would be a lot of work and maintenance to have interactive code example in all languages. I'd rather just put in a disclaimer or change it to pseudo-code, because the purpose is to server as an example for understanding, not for copy-paste.

Reply:
I will add I don't think people really use indexes like are outlined in this document. It's very fragile to keep all of these different indexes in sync, and if it breaks the system is extremely hard to recover from.

Reply:
Do you want to delete this page?

I didn't consider that. I just tried to fix Kyle's comments.

Reply:
@madolson if you want to delete the indexing page, I suggest we do that in another PR. We'd also need to remove links to it from other pages, etc.